mono/eglib/goutput.c

```diff
@@ -126,12 +126,23 @@ g_log (const gchar *log_domain, GLogLevelFlags log_level, const gchar *format, .
 	va_end (args);
 }
 
+static char *failure_assertion = NULL;
+
+const char *
+g_get_assertion_message (void)
+{
+	return failure_assertion;
+}
+
 void
 g_assertion_message (const gchar *format, ...)
 {
 	va_list args;
 
 	va_start (args, format);
+
+	g_vasprintf (&failure_assertion, format, args);
```

Comment:
This breaks assert. It gets varargs after g_vasprintf has bumped it past the parameters.
You get garbage or a crash in the assertion failure.
Put va_start after it, or g_vasprintf into a local and use it for both (g_log("%s", local)).

Reply:
Adding va_start is small change, reusing sprintf output is "faster", and maybe feels safer -- is reusing varargs portable with va_start?

Reply:
https://github.com/mono/mono/pull/8727/commits/a1baf487d073f6d66d11ac0134f2fa6b0e06408e fixes this.